### PR TITLE
Add notices for development builds

### PR DESCRIPTION
**Pull request by SadPencil**
This PR adds a "(Development Build)" suffix in title bar, as well as a log message, for non-release builds.

A build is considered as a release build if all of the following conditions are met:
- The branch name is "master"
- The commit has a tag with a correct version number recognized by GitVersion

Also, this PR adds "rc" as the tag suffix for non-release builds in master branch.

Also, **this PR changes the version increment assumption**: Patch version instead of Minor version will be increased even for develop branch. Since we aren't actually following SemVer.

Note: metadata of DLL/EXE files are not modified due to technical difficulties. Feel free to draft another PR if someone can achieve it.

图片

~~Since `ThisAssembly.Git` package is introduced, one needs to restart Visual Studio after a successful build **for the first time**. (We already have to do so since translation stub was introduced)~~

### CODE REVIEW

**Comment by MahBoiDeveloper** [comment]
Artifact for TSC works as intended.

image

**Comment by MahBoiDeveloper** [comment]
Self-builded binaries also works.

image

File: DXMainClient/DXGUI/GameClass.cs [hunk 1]
```diff
@@ -69,6 +69,10 @@ protected override void Initialize()
             Window.Title = string.IsNullOrEmpty(windowTitle) ?
                 string.Format("{0} Client", MainClientConstants.GAME_NAME_SHORT) : windowTitle;
 
+#if DEVELOPMENT_BUILD
+            Window.Title += " (Development Build)";
```
Comment (MahBoiDeveloper) [inline DXMainClient/DXGUI/GameClass.cs:73]: Please, add L10N e.g., ` Window.Title += " " + "(Development Build)".L10N("Client:Main:DeveopmentBuildTitle");`

**Review by MahBoiDeveloper — CHANGES_REQUESTED** [review]

File: DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs [hunk 2]
```diff
@@ -556,10 +556,23 @@ private void PostUIInit()
...
+            clientVersion = $"{ThisAssembly.Git.Branch}@{ThisAssembly.Git.Sha}";
+#endif
+
             connectionManager.MainChannel.AddMessage(new ChatMessage(Color.White, Renderer.GetSafeString(
-                    string.Format("*** DTA CnCNet Client version {0} ***".L10N("Client:Main:CnCNetClientVersionMessage"), Assembly.GetAssembly(typeof(CnCNetLobby)).GetName().Version),
+                    string.Format("*** DTA CnCNet Client version {0} ***".L10N("Client:Main:CnCNetClientVersionMessage"), clientVersion),
                     lbChatMessages.FontIndex)));
 
+            {
+                string developBuildWarningMessage = "This is a development build. Stability and reliability may not be fully guaranteed.".L10N("Client:Main:DevelopmentBuildWarning");
```
Comment (Metadorius) [inline DXMainClient/DXGUI/Multiplayer/CnCNet/CnCNetLobby.cs:569]:
```suggestion
                string developBuildWarningMessage = "This is a development build of the client. Stability and reliability may not be fully guaranteed.".L10N("Client:Main:DevelopmentBuildWarning");
```

**Comment by SadPencil** [comment]
@MahBoiDeveloper @Metadorius review comments addressed. Request re-review

**Comment by MahBoiDeveloper** [comment]
All text are translatable.

Char warning message:
image

Title:
image

INI key works as intended:
image

**Comment by MahBoiDeveloper** [comment]
Please, add this key with default value and description into https://github.com/CnCNet/xna-cncnet-client/blob/develop/DXMainClient/Resources/ClientDefinitions.ini

**Comment by MahBoiDeveloper** [comment]
> Please, add this key with default value and description into https://github.com/CnCNet/xna-cncnet-client/blob/develop/DXMainClient/Resources/ClientDefinitions.ini

@SadPencil merge https://github.com/SadPencil/xna-cncnet-client/pull/6 into this branch please.

File: DXMainClient/DXGUI/GameClass.cs [hunk 3]
```diff
@@ -69,6 +69,15 @@ protected override void Initialize()
             Window.Title = string.IsNullOrEmpty(windowTitle) ?
                 string.Format("{0} Client", MainClientConstants.GAME_NAME_SHORT) : windowTitle;
 
+            {
+                string developBuildTitle = "(Development Build)".L10N("Client:Main:DevelopmentBuildTitle");
+
+#if DEVELOPMENT_BUILD
+                if (ClientConfiguration.Instance.ShowDevelopmentBuildWarnings)
+                    Window.Title += " " + developBuildTitle;
+#endif
```
Comment (Metadorius) [inline DXMainClient/DXGUI/GameClass.cs:78]: Is it needed to translate parentheses? Maybe
```suggestion
                string developBuildTitle = "Development Build".L10N("Client:Main:DevelopmentBuildTitle");

#if DEVELOPMENT_BUILD
                if (ClientConfiguration.Instance.ShowDevelopmentBuildWarnings)
                    Window.Title += $" ({developBuildTitle})";
#endif
```

would work better?

**Review by Metadorius — CHANGES_REQUESTED** [review]
It's also needed to write different file metadata for non-release builds. From what I understand currently the metadata is not altered.

**Comment by SadPencil** [comment]
> It's also needed to write different file metadata for non-release builds. From what I understand currently the metadata is not altered.

Don't know way to achieve this

**Comment by MahBoiDeveloper** [comment]
> > It's also needed to write different file metadata for non-release builds. From what I understand currently the metadata is not altered.
> 
> Don't know way to achieve this

Isn't already done?
image

**Comment by Metadorius** [comment]
Eh that works too I guess. I would have preferred complete wipe of course. Maybe @frg2089 or @devo1929 could help

**Pull request merged**